test/commands.test.js

```diff
@@ -435,6 +435,33 @@ test('[commands.commandContext] iterates through operations', function(assert) {
   context.next();
 });
 
+test('[commands.commandContext] callback with diffs', function(assert) {
+  var ops = [
+    commands.operations.confirmParameters,
+    commands.operations.confirmTemplate
+  ];
+
+  sinon.stub(prompt, 'confirm').yields(null, true);
+
+  var context = commands.commandContext(opts, 'testing', ops, function(err, performed, diffs) {
+    assert.ifError(err, 'success');
+    assert.equal(performed, true, 'the requested command was performed');
+    assert.deepEqual(diffs, {
+      parameters: ' {\n\u001b[32m+  newones: "too"\u001b[39m\n }\n',
+      template: '\u001b[90m {\n\u001b[39m\u001b[31m-  "old": "template"\n\u001b[39m\u001b[32m+  "new": "template"\n\u001b[39m\u001b[90m }\u001b[39m'
+    }, 'callback provides diffs as 3rd arg');
```

Comment:
can you explain what is happening here wrt the unicode characters?

Reply:
I think I understood that these are all related to how jsonDiff.diffString() works, and have been modified to be diff strings, so the unicode characters are to recreate something like:

<img width="152" alt="screen shot 2018-02-20 at 9 05 32 pm" src="https://user-images.githubusercontent.com/3166852/36433136-d24d55ae-1681-11e8-9cc6-da5fa48ee5df.png">

Reply:
The unicode characters are instructions to color the text green or red or gray in the terminal.